mindwalc/datastructures.py

```diff
@@ -211,4 +214,110 @@ def node_count(self):
             left_count = self.left.node_count
         if self.right is not None:
             right_count = self.right.node_count
-        return 1 + left_count + right_count
\ No newline at end of file
+        return 1 + left_count + right_count
+
+    def visualize(self, output_path, _view=True, node_properties_function=None, meta_infos='', as_pdf=True):
+        """
+        Visualise the tree with [graphviz](http://www.graphviz.org/) and save it as pdf and or .gv file.
+
+        **Params**
+        ----------
+          - `output_path` (string) - where the file needs to be saved
+          - `_view` (boolean) - open the pdf after generation or not
+          - `node_properties_function` (function) - function (params: xxx). Passed function returns a
+            dict which contains the properties of the generated graph,
+            using the Graphviz dot languade. see self._default_tree_visualization as example!
+          - `meta_infos` (string) - a string, usually containing some usefully infos about the tree
+            (which params used for decision tree classification training, which dataset used ...)
+          - `as_pdf` (boolean) - if True, the output file will be a pdf % a .gv file, otherwise only one .gv file
+        **Returns**
+        -----------
+            nothing
+        """
+
```

Comment:
This is where I would import graphviz. This is the only public function so it should be enough here.